App.js

```diff
@@ -32,6 +34,10 @@ const App = () => {
     const subscription = AppState.addEventListener('change', (nextAppState) => {
       if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
         auth();
+
+        // saved data in the sue module is deleted if the application returns to the foreground from
+        // the background or inactive state
+        AsyncStorage.removeItem(SUE_REPORT_VALUES);
```

Comment:
i guess t should only happen if the app was closed entirely, not when just moved in the background. imagine you start filling the form, receive a short message, switches to messaging app and return after that. in that case the form data should be still present

Reply:
But we can't see if the application has been closed completely. Then I'll have to find another way.  🤷‍♂️

Reply:
I added the code to the section where we make SplashScreen hidden. I think it is much better in terms of code structure and logic.

https://github.com/smart-village-solutions/smart-village-app-app/commit/1d11ef5dd20fd0deb67b20f265b923b31b0d30f8